src/hcf-release/src/acceptance-tests-brain/test-scripts/008_cf_usb_test.sh

```diff
@@ -0,0 +1,200 @@
+#!/bin/bash
+# # ## ### ##### ######## #############  #####################
+## Explanations and overview:
+
+## This test does a lot of setup to determine that the universal
+## service broker (cf-usb) is actually working.
+#
+## 0. A domain for TCP routing is created, to connect all the pieces.
+#
+## 1. A local mysql server is started as an app and made available
+##    through cf TCP routing.
+#
+## 2. The mysql sidecar is started as an app, configured to talk to
+##    the mysql server from (1).
+#
+## 3. The cf-usb is configured to talk to and use the side car.
+#
+## 4. Then we can check that mysql appears in the marketplace, create
+##    a service from it, and check that this service is viewable too.
+#
+## In the code below these phases are marked with "--(N)--" where N is
+## the step number.
+#
+## Note, the applications of step 1 and 2 are docker apps. This is why
+## the `pre-start.sh` script enables the `diego_docker` feature-flag
+## of CF. For step (3) the `pre-start.sh` script extended the `cf`
+## client with the `cf-usb-plugin` plugin.
+
+function get_port
+{
+    file="${1}"
+    port="$(awk '/Route .* has been created/ {print $2}' < "${TMP}/${file}" | cut -f 2 -d ':')"
+    if [ -z "${port}" ]; then
+	echo 1>&2 "ERROR: Could not determine the assigned random port number for $1"
+	echo 1>&2 "ERROR: Mapping route to random port failed for $1"
+	exit 1
+    fi
+    echo "${port}"
+}
+
+function wait_on_port
+{
+    endpoint="${CF_TCP_DOMAIN}:${1}"
+    for (( i = 0; i < 12 ; i++ )) ; do
+	if curl --fail -s -o /dev/null "${endpoint}" ; then
+            break
+	fi
+	sleep 5
+    done
+    # Last try, any error will abort the test
+    curl "${endpoint}"
```

Comment:
```
+ curl -v tcp.cf-dev.io:20003
* Rebuilt URL to: tcp.cf-dev.io:20003/
* Hostname was NOT found in DNS cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 192.168.77.77...
* Connected to tcp.cf-dev.io (192.168.77.77) port 20003 (#0)
> GET / HTTP/1.1
> User-Agent: curl/7.37.0   
> Host: tcp.cf-dev.io:20003 
> Accept: */*
>
{ [data not shown]
100   104    0   104    0     0  20858      0 --:--:-- --:--:-- --:--:-- 26000
* Connection #0 to host tcp.cf-dev.io left intact
5.7.20bws)s=
���_;   J5[dd0mysql_native_password��Got packets out of order
```
So yeah, we're talking HTTP at the MySQL port and it's… somehow succeeding?

Reply:
Hm. The only thing I want out of this is to wait until the mysql server port is available, i.e. that we can connect to it, before progressing with the setup and testing.

Reply:
Close enough. I would have used `nc` or `head -c0 </dev/tcp/host/port`, but if it works, why not.

Reply:
That last is interesting, 1st time I see that. Yeah, forgot about `nc`. We can make it a cleanup ticket if you see the need.